### PR TITLE
Make make help colorful and simple

### PR DESCRIPTION
**Pull request by szepeviktor**
Colorful and simple.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,8 +38,8 @@ BLACKFIRE = blackfire
 
 .PHONY: help
 help:
-	@echo "\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n"
-	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//' | awk 'BEGIN {FS = ":"}; {printf "\033[33m%s:\033[0m%s\n", $$1, $$2}'
+	@echo -e "\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n"
```
Comment (szepeviktor) [inline Makefile:41]: The `-e` option was missing.

File: Makefile [hunk 2]
```diff
@@ -38,8 +38,8 @@ BLACKFIRE = blackfire
...
+	@sed -n -e 's|^\(.\+:\) ##\(.\+\)$$|\o033[33m\1 \o033[0m\2|p' $(MAKEFILE_LIST)
```
Comment (szepeviktor) [inline Makefile:42]: One `sed` does it all.

**Comment by szepeviktor** [comment]
Why does  fail?
On my machine
```
$ make help
\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n
```

**Comment by theofidry** [comment]
The result after running `make help` before/after the changes of your PR for me:

Screenshot 2023-11-05 at 22 42 55

Screenshot 2023-11-05 at 22 43 04

**Comment by szepeviktor** [comment]
Is your shell Bash?

**Comment by theofidry** [comment]
zsh*

**Comment by szepeviktor** [comment]
I see.
`echo` is not portable. https://unix.stackexchange.com/a/700679

**Comment by theofidry** [comment]
I indeed use this form in another project https://github.com/theofidry/makefile/blob/main/Makefile#L42.

Note I also didn't use sed, because it behaves different on OSX which is super annoying. And actually I think it's the culprit for the failure. The CI works fine as it's ubuntu, but locally I get the same as the "after" screenshot from before.

**Comment by szepeviktor** [comment]
> Note I also didn't use sed, because it behaves different on OSX

Yes. GNU sed has an option: `sed --posix` to disable GNU extensions.
Indeed it does not work with that option!

**Comment by theofidry** [comment]
TIL

**Comment by szepeviktor** [comment]
The "one or more" `+` sign is the culprit.

I give up. Sorry.

**Comment by theofidry** [comment]
So I think if you revert just the `@sed` change it's still a mergeable change :) I would be happy to simplify the current grep command but I admit I also didn't care enough to try to figure a way to do so (it really didn't look simple and the sort of issues I could spend hours on)

**Comment by theofidry** [comment]
>I give up. Sorry.

No worries, it's a nasty one

**Comment by szepeviktor** [comment]
> it really didn't look simple

Compatibility is a hot topic!
I keep investigating in my secret kitchen ...

**Pull request closed without merging**